### ISSUE TITLE
Support for cross_validation

### ISSUE THREAD

**Issue by oasis789** [dask/dask-ml#165]
Hi,

Is there any reason why cross_validation has not been implemented with a dask alternative? 

This would be very useful for me and others I can imagine.

**Comment by TomAugspurger** [comment]
Probably just lack of time:) Which bits of cross validation do you need?

I have a branch at https://github.com/TomAugspurger/dask-ml/tree/train-test-split that started train-test-split, but haven't had a chance to finish it off. If train_test_split is what you need, would you be interested in pushing it through to the end?

**Comment by oasis789** [comment]
I'm after a dask version of `sklearn.model_selection.cross_validate` that can use `sklearn.model_selection.ShuffleSplit` and `sklearn.model_selection.learning_curve`.

I imagine these will be useful for other people also since running these in a distributed setup can save a heck of a lot of time. I'll have a look at what you have done and try and take some time out to work on this.

**Comment by TomAugspurger** [comment]
Are you working with dask or NumPy arrays? IOW, is your data larger than memory?

`sklearn.model_selection.cross_validate` and `learning_curve` should work already in a `with joblib.backend('dask')` context manager for in-memory datasets, though I haven't tested it.

For dask arrays, we'll have a bit of work to do.

**Comment by oasis789** [comment]
My data isn't in dask arrays so that shouldn't be a problem.

I'll have to try the context manager and see what happens. Although, I have parallelised predictions in my classifier so I'm not sure if that will cause problems.

**Comment by TomAugspurger** [comment]
Thanks, let me know how it goes.

**Comment by TomAugspurger** [comment]
@oasis789 I added a screencast to the the docs http://dask-ml.readthedocs.io/en/latest/joblib.html (https://youtu.be/5Zf6DQaf7jk)

This shows using GridSearchCV, which will be using ShuffleSplit under the hood. Everything seems to be working fine, but let me know if not.
